SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/EventTracingTests/EventTracingTest.cpp

```diff
@@ -128,29 +150,31 @@ void AEventTracingTest::GatherData()
 		return A.CreationTime > B.CreationTime;
 	});
 
-	bool bFoundClient = false;
-	bool bFoundWorker = false;
+	FPlatformProcess::Sleep(1); // Worker bug means file may not be flushed by the OS (WRK-2396)
+
+	int FoundClient = 0;
+	int FoundWorker = 0;
 	for (const FileCreationTime& FileCreation : FileCreationTimes)
 	{
-		if (!bFoundClient && FileCreation.FilePath.Contains("UnrealClient"))
+		if (FoundClient != 1 && FileCreation.FilePath.Contains("UnrealClient"))
 		{
 			GatherDataFromFile(FileCreation.FilePath);
-			bFoundClient = true;
+			FoundClient++;
 		}
 
-		if (!bFoundWorker && FileCreation.FilePath.Contains("UnrealWorker"))
+		if (FoundWorker != 2 && FileCreation.FilePath.Contains("UnrealWorker"))
 		{
 			GatherDataFromFile(FileCreation.FilePath);
-			bFoundWorker = true;
+			FoundWorker++;
 		}
 
-		if (bFoundClient && bFoundWorker)
+		if (FoundClient == 1 && FoundWorker == 2)
 		{
 			break;
 		}
 	}
 
-	if (!bFoundClient || !bFoundWorker)
+	if (FoundClient != 1 || FoundWorker != 2)
```

Comment:
This suggests we need to run in multi worker. That should not have to be the case.

Reply:
Ah, I have adjusted the map to run with 2_1 zoning. Think I should add another map? It's nice to confirm that single client tracing does also function correctly with zoning but maybe it's a test for another place. Maybe a zoning map with the same tests in + the cross-server stuff?